lib/system/extra.ex

```diff
@@ -0,0 +1,84 @@
+defmodule System.Extra do
+  @moduledoc """
+  Extensions to the standard library's System module
+
+  """
+
+  alias __MODULE__.CmdStreamer
+
+
+
+  #########################################################
+  # Types
+  #########################################################
+
+  @typep acc :: any
+  @typep element :: any
+
+
+
+  #########################################################
+  # Public API
+  #########################################################
+
+  @doc """
+  Asserts that the provided path is valid and an executable binary.
+
+  iex> System.Extra.assert_executable!("/usr/local/bin/mix")
+  :ok
+
+  """
+  @spec assert_executable!(Stream.t) :: :ok | no_return
+  def assert_executable!(path) do
+    case System.find_executable(path) do
+      result when is_binary(result) ->
+        :ok
+
+      nil ->
+        raise(File.Error, reason: :enoent, action: "find executable", path: path)
+    end
+  end
+
+
+  @doc """
+  Streams output from a command by line.
+
+  Will raise an error if the command exits with a non-zero exit code.
+
+  Will also error on lines over 20,000 characters long. Don't be that guy.
```

Comment:
Why would it error on lines over 20,000 characters? That seems an odd limit for a generic library

Reply:
In fact that actually seems dangerous. Plenty of xml files or streams can be output with over 20 kilobytes of data in a single line.